system/pages/highscores.php

```diff
@@ -209,158 +220,103 @@
+		$player['online'] = (isset($is_online[$player['id']]) ? 1 : 0);
+	} else {
+		if(!isset($player['online'])) {
+			$player['online'] = 0;
+		}
+	}
+
+	if(++$i <= $configHighscoresPerPage)
 	{
 		if($skill == POT::SKILL__MAGIC)
 			$player['value'] = $player['maglevel'];
-		else if($skill == POT::SKILL__LEVEL)
+		else if($skill == POT::SKILL__LEVEL) {
 			$player['value'] = $player['level'];
-echo '
-		<tr bgcolor="' . getStyle($i) . '">';
-		if($config['account_country'])
-			echo '<td>' . getFlagImage($player['country']) . '</td>';
-echo '
-			<td>' . ($offset + $i) . '.</td>';
-			if($config['highscores_outfit'])
-			echo '<td><img style="position:absolute;margin-top:' . (in_array($player['looktype'], array(75, 266, 302)) ? '-15px;margin-left:5px' : '-45px;margin-left:-25px') . ';" src="' . $config['outfit_images_url'] . '?id=' . $player['looktype'] . ($outfit_addons ? '&addons=' . $player['lookaddons'] : '') . '&head=' . $player['lookhead'] . '&body=' . $player['lookbody'] . '&legs=' . $player['looklegs'] . '&feet=' . $player['lookfeet'] . '" alt="" /></td>';
-
-echo '
-			<td>
-				<a href="' . getPlayerLink($player['name'], false) . '">
-					<span style="color: ' . ($player['online'] > 0 ? 'green' : 'red') . '">' . $player['name'] . '</span>
-				</a>';
-				if($config['highscores_vocation']) {
-					if(isset($player['promotion'])) {
-						if((int)$player['promotion'] > 0)
-							$player['vocation'] += ($player['promotion'] * $config['vocations_amount']);
-					}
-
-					$tmp = 'Unknown';
-					if(isset($config['vocations'][$player['vocation']])) {
-						$tmp = $config['vocations'][$player['vocation']];
-					}
-					echo '<br/><small>' . $tmp . '</small>';
+			$player['experience'] = number_format($player['experience']);
+		}
+
+		if($configHighscoresVocation) {
+			if(isset($player['promotion'])) {
+				if((int)$player['promotion'] > 0) {
+					$player['vocation'] += ($player['promotion'] * $configVocationsAmount);
 				}
-echo '
-			</td>
-			<td>
-				<div style="text-align:center">'.$player['value'].'</div>
-			</td>';
+			}
+
+			$tmp = 'Unknown';
+			if(isset($configVocations[$player['vocation']])) {
+				$tmp = $configVocations[$player['vocation']];
+			}
+
+			$player['vocation'] = $tmp;
+		}
 
-		if($skill == POT::SKILL__LEVEL)
-			echo '<td><div style="text-align:center">' . number_format($player['experience']) . '</div></td>';
 
-		echo '</tr>';
+		$player['link'] = getPlayerLink($player['name'], false);
+		$player['flag'] = getFlagImage($player['country']);
+		$player['outfit'] = '<img style="position:absolute;margin-top:' . (in_array($player['looktype'], array(75, 266, 302)) ? '-15px;margin-left:5px' : '-45px;margin-left:-25px') . ';" src="' . config('outfit_images_url') . '?id=' . $player['looktype'] . ($outfit_addons ? '&addons=' . $player['lookaddons'] : '') . '&head=' . $player['lookhead'] . '&body=' . $player['lookbody'] . '&legs=' . $player['looklegs'] . '&feet=' . $player['lookfeet'] . '" alt="" />';
```

Comment:
There is bug with undefined index: looktype, lookhead etc. when highscores_outfit is false in config.php

```suggestion
		if($configHighscoresOutfit) {
			$player['outfit'] = '<img style="position:absolute;margin-top:' . (in_array($player['looktype'], array(75, 266, 302)) ? '-15px;margin-left:5px' : '-45px;margin-left:-25px') . ';" src="' . config('outfit_images_url') . '?id=' . $player['looktype'] . ($outfit_addons ? '&addons=' . $player['lookaddons'] : '') . '&head=' . $player['lookhead'] . '&body=' . $player['lookbody'] . '&legs=' . $player['looklegs'] . '&feet=' . $player['lookfeet'] . '" alt="" />';
		}
```